sbt-scalajs-bundler/src/main/scala/scalajsbundler/NpmPackage.scala

```diff
@@ -0,0 +1,40 @@
+package scalajsbundler
+
+import java.io.FileReader
+import org.scalajs.core.tools.json
+import org.scalajs.core.tools.json.{JSON, JSONDeserializer, JSONObjExtractor}
+import sbt._
+
+case class NpmPackage(version: String) {
+  def major: Option[Int] = {
+    val r = """^(\d+)(\..*|)$""".r
+    r findFirstMatchIn version flatMap { case r(v, x) =>
+      try {
+        Some(v.toInt)
+      } catch {
+        case e: Exception => None
+      }
+    }
+  }
```

Comment:
I think you could rewrite this as follows:

~~~ scala
version match {
  case r(v, _) => Try(v.toInt).toOption
  case _ => None
}
~~~